test/e2e/metrics_test.go

```diff
@@ -156,11 +159,43 @@ func (c *MetricsTestConfig) validate(token string) {
 	require.Contains(c.t, string(output), "200 OK", "Metrics endpoint did not return 200 OK")
 }
 
-// cleanup created resources
+// cleanup removes the created resources. Uses a context with timeout to prevent hangs.
 func (c *MetricsTestConfig) cleanup() {
 	c.t.Log("Cleaning up resources")
 	_ = exec.Command(c.client, "delete", "clusterrolebinding", c.clusterBinding, "--ignore-not-found=true").Run()
 	_ = exec.Command(c.client, "delete", "pod", c.curlPodName, "-n", c.namespace, "--ignore-not-found=true").Run()
+
+	// Create a context with a 60-second timeout.
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+
+	// Wait for the ClusterRoleBinding to be deleted.
+	if err := waitForDeletion(ctx, c.client, "clusterrolebinding", c.clusterBinding); err != nil {
+		c.t.Logf("Error waiting for clusterrolebinding deletion: %v", err)
+	} else {
+		c.t.Log("ClusterRoleBinding deleted")
+	}
+
+	// Wait for the Pod to be deleted.
+	if err := waitForDeletion(ctx, c.client, "pod", c.curlPodName, "-n", c.namespace); err != nil {
+		c.t.Logf("Error waiting for pod deletion: %v", err)
+	} else {
+		c.t.Log("Pod deleted")
+	}
+}
+
+// waitForDeletion uses "kubectl wait" to block until the specified resource is deleted
+// or until the 30-second timeout is reached.
+func waitForDeletion(ctx context.Context, client, resourceType, resourceName string, extraArgs ...string) error {
+	args := []string{"wait", "--for=delete", resourceType, resourceName}
+	args = append(args, extraArgs...)
+	args = append(args, "--timeout=30s")
```

Comment:
Since the default timeout for a kubectl delete --force command is 30 seconds, I don’t think wait here just 30 seconds is enough. Could we change it for 1 minute since mainly every place on the tests we give the grace of 1 minute?

Reply:
done